src/payments-processors/lnurl-payments-processor.ts

```diff
@@ -0,0 +1,69 @@
+import { AxiosInstance } from 'axios'
+import { Factory } from '../@types/base'
+
+import { CreateInvoiceRequest, GetInvoiceResponse, IPaymentsProcessor } from '../@types/clients'
+import { Invoice, InvoiceStatus, InvoiceUnit } from '../@types/invoice'
+import { createLogger } from '../factories/logger-factory'
+import { randomUUID } from 'crypto'
+import { Settings } from '../@types/settings'
+
+const debug = createLogger('alby-payments-processor')
+
+export class LnurlPaymentsProcesor implements IPaymentsProcessor {
+  public constructor(
+    private httpClient: AxiosInstance,
+    private settings: Factory<Settings>
+  ) {}
+
+  public async getInvoice(invoice: Invoice): Promise<GetInvoiceResponse> {
+    debug('get invoice: %s', invoice.id)
+
+    try {
+      const response = await this.httpClient.get(invoice.verifyURL)
+
+      return {
+        id: invoice.id,
+        status: response.data.settled ? InvoiceStatus['COMPLETED'] : InvoiceStatus['PENDING'],
+      }
+    } catch (error) {
+      console.error(`Unable to get invoice ${invoice.id}. Reason:`, error)
+
+      throw error
+    }
+  }
+
+  public async createInvoice(request: CreateInvoiceRequest): Promise<any> {
+    debug('create invoice: %o', request)
+    const {
+      amount: amountMsats,
+      description,
+      requestId,
+    } = request
+
+    try {
+      const response = await this.httpClient.get(`${this.settings().paymentsProcessors?.lnurl?.invoiceURL}/callback?amount=${amountMsats}&comment=${requestId}`)
+
+      const result = {
+        id: randomUUID(),
```

Comment:
although this works, I'm concerned we won't be able to look up this invoice on your LNURL provider. Should we instead use UUID v5 which uses namespaces and can be used to generate deterministic UUIDs? That way we can pass the invoice description and always get the same UUID back.

Reply:
We didn't have a uuid string coming from the response: https://getalby.com/lnurlp/adithya/callback?amount=10000
So we thought of using a random UUID.

> instead use UUID v5 which uses namespaces and can be used to generate deterministic UUIDs?

The problem is that we don't return the description (and because of that I changed the getInvoice as well to take the whole invoice instead of just invoice.id)

Reply:
well, this could be a future improvement, not a blocker. But we definitely want to give relay operators the ability to go to their LNURL provider to lookup invoices. If they can at least see descriptions which should include the kind of fee and pubkey then it's all good for now.